nvflare/fuel/utils/pipe/pipe_handler.py

```diff
@@ -149,53 +223,70 @@ def _make_event_message(topic: str, data):
     def wait_for_other_end(self, timeout: Optional[float] = None):
```

Comment:
What is this method for? Are you imagining that the caller can wait for the peer to come up?

Reply:
In the LauncherExecutor case,

There is a situation where the 3rd party is launched by subprocess.Popen,
then it takes a long time for it to get to the PipeHandler.start (data loading, model setup takes a long time)

In that scenario, if the LauncherExecutor sends the task right away and wait for "peer_read_timeout", it can easily times out. (since the other end is still setting up data, model)

I add this so that the LauncherExecutor will only sends the task AFTER the other side's PipeHandler is up and running.

Reply:
I see. Make sense.